esp_flasher/own_esptool.py

```
@@ -78,6 +79,53 @@
+            stub["text"] = base64.b64decode(stub["text"])
+            if "data" in stub:
+                stub["data"] = base64.b64decode(stub["data"])
+            else:
+                stub["data"] = b""
+                stub.setdefault("data_start", 0)
+            self._cached_stub = stub
+            return stub
+        except (IOError, OSError, json.JSONDecodeError, KeyError) as e:
+            raise FatalError(
+                f"Failed to load stub code for {self.chip_name}: {e}"
+            )
```

Ruff (0.15.7)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)